packages/parser/src/parser.ts

```diff
@@ -164,10 +164,14 @@ function parseForESLint(
         );
       }
     }
-    if (compilerOptions.emitDecoratorMetadata === true) {
-      analyzeOptions.emitDecoratorMetadata =
-        compilerOptions.emitDecoratorMetadata;
-    }
+  }
+
+  const emitDecoratorMetadata = services.hasFullTypeInformation
+    ? services.program.getCompilerOptions().emitDecoratorMetadata === true
+    : options.emitDecoratorMetadata === true;
```

Comment:
instead of duplicating the check and duplicating the `getCompilerOptions` call - we should split this up.

i.e. something like this:

```ts
let emitDecoratorMetadata = options.emitDecoratorMetadata === true;
if (services.hasFullTypeInformation) {
  // ......
  emitDecoratorMetadata = compilerOptions.emitDecoratorMetadata
}
```

Reply:
I see. Trying to refactor